cirq/experiments/fidelity_estimation.py

```diff
@@ -470,17 +505,160 @@ def __enter__(
+    are extracted from the lists of circuits and used to define one pair. Instead of using
+    `combinations` to shuffle the circuits for each pair, we just use each circuit (in order)
+    for the one pair.
+    """
+    q0, q1 = _verify_and_get_two_qubits_from_circuits(circuits)
+    circuits = [
+        circuit.transform_qubits(lambda q: {q0: devices.LineQubit(0), q1: devices.LineQubit(1)}[q])
+        for circuit in circuits
+    ]
+    return [
+        CircuitLibraryCombination(
+            layer=None,
+            combinations=np.arange(len(circuits))[:, np.newaxis],
+            pairs=[(q0, q1)],
+        )
+    ], circuits
+
+
+def _zip_circuits(
+    circuits: Sequence['cirq.Circuit'], combinations_by_layer: List[CircuitLibraryCombination]
+) -> List[_ZippedCircuit]:
+    """Helper function used in `sample_2q_xeb_circuits` to zip together circuits.
+
+    This takes a sequence of narrow `circuits` and "zips" them together according to the
+    combinations in `combinations_by_layer`.
+    """
+
+    # Check `combinations_by_layer` is compatible with `circuits`.
+    for layer_combinations in combinations_by_layer:
+        if np.any(layer_combinations.combinations < 0) or np.any(
+            layer_combinations.combinations >= len(circuits)
+        ):
+            raise ValueError("`combinations_by_layer` has invalid indices.")
+
+    zipped_circuits: List[_ZippedCircuit] = []
+    for layer_i, layer_combinations in enumerate(combinations_by_layer):
+        for combination_i, combination in enumerate(layer_combinations.combinations):
+            wide_circuit = Circuit.zip(
+                *(
+                    circuits[i].transform_qubits(lambda q: pair[q.x])
+                    for i, pair in zip(combination, layer_combinations.pairs)
+                )
+            )
+            zipped_circuits.append(
+                _ZippedCircuit(
+                    wide_circuit=wide_circuit,
+                    pairs=layer_combinations.pairs,
+                    combination=combination.tolist(),
+                    layer_i=layer_i,
+                    combination_i=combination_i,
+                )
+            )
+    return zipped_circuits
+
+
+def _generate_sample_2q_xeb_tasks(
+    zipped_circuits: List[_ZippedCircuit], cycle_depths: Sequence[int]
+) -> List[_Sample2qXEBTask]:
+    """Helper function used in `sample_2q_xeb_circuits` to prepare circuits in sampling tasks."""
+    tasks: List[_Sample2qXEBTask] = []
+    for cycle_depth in cycle_depths:
+        for zipped_circuit in zipped_circuits:
+            circuit_depth = cycle_depth * 2 + 1
+            assert circuit_depth <= len(zipped_circuit.wide_circuit)
+            # Slicing creates a copy, although this isn't documented
+            prepared_circuit = zipped_circuit.wide_circuit[:circuit_depth]
+            for pair_i, pair in enumerate(zipped_circuit.pairs):
+                prepared_circuit += ops.measure(*pair, key=str(pair_i))
```

Comment:
Have you checked if this aligns all the measurements in one moment?

Reply:
for the current code, it should just because these are dense circuits by construction; but I've changed it to add the measure operations in one moment. Good spot!